drone-fly-app/src/main/java/com/expediagroup/dataplatform/dronefly/app/context/CommonBeans.java

```diff
@@ -54,6 +59,13 @@ public HiveConf hiveConf() {
     return new HiveConf();
   }
 
+  @Bean
+  @Primary
```

Comment:
Why @primary?
@ConfigurationProperties maybe you can set the prefix here so only those properties are mapped into the Properties object.

Reply:
Primary: spring boot autoconfigures a Properties bean that contains only system variables, for getting system and env variables we need to define this bean, and set it as primary to avoid clashing.

I changed the validation in code to the configuration properties prefix as mentioned, thanks.

Reply:
ah there is some internal spring bean got it. ok